labs_api/cache.py

```diff
@@ -0,0 +1,100 @@
+from datetime import datetime, timedelta
+from typing import Optional, TypeVar
+
+from prediction_market_agent_tooling.gtypes import HexAddress
+from prediction_market_agent_tooling.loggers import logger
+from prediction_market_agent_tooling.tools.utils import utcnow
+from pydantic import BaseModel
+from sqlmodel import Field, Session, SQLModel, create_engine, desc, select
+
+from labs_api.config import Config
+
+
+class ResponseCacheModel(BaseModel):
+    market_id: HexAddress
+    created_at: datetime
+
+
+class ResponseCacheSQLModel(SQLModel):
+    __table_args__ = {"extend_existing": True}
+    id: Optional[int] = Field(default=None, primary_key=True)
+    market_id: str = Field(index=True)
+    datetime_: datetime = Field(index=True)
+    json_dump: str
+
+
+from typing import Generic, TypeVar
+
+ResponseCacheModelVar = TypeVar("ResponseCacheModelVar", bound=ResponseCacheModel)
+ResponseCacheSQLModelVar = TypeVar(
+    "ResponseCacheSQLModelVar", bound=ResponseCacheSQLModel
+)
+
+
+class ResponseCache(Generic[ResponseCacheModelVar, ResponseCacheSQLModelVar]):
```

Comment:
This is 100% reusable by both insights and invalid endpoint, so I made it a generic